packages/abi-coder/src/encoding/coders/BigNumberCoder.test.ts

```diff
@@ -23,6 +23,44 @@ describe('BigNumberCoder', () => {
     expect(actual).toStrictEqual(expected);
   });
 
+  it('should encode a u64 [max safe integer]', () => {
+    const coder = new BigNumberCoder('u64');
+    const value: number = Number.MAX_SAFE_INTEGER;
+    const expected = new Uint8Array([0, 31, 255, 255, 255, 255, 255, 255]);
+
+    const data = coder.encode(value);
+
+    expect(data).toEqual(expected);
+  });
+
+  it('should throw an error when encoding [number more than max safe integer]', () => {
+    const coder = new BigNumberCoder('u64');
+    const value: number = Number.MAX_SAFE_INTEGER + 1;
+
+    expect(() => coder.encode(value)).toThrow(
+      new FuelError(ErrorCode.ENCODE_ERROR, 'Invalid u64 type - number value is too large.')
+    );
+  });
+
+  it('should encode a u64 [very big number - as string]', () => {
+    const coder = new BigNumberCoder('u64');
+    const value: string = '76472027892439376';
+    const expected = new Uint8Array([1, 15, 174, 231, 121, 200, 89, 80]);
+
+    const data = coder.encode(value);
+
+    expect(data).toEqual(expected);
+  });
+
+  it('should throw an error when encoding [number more than max safe integer]', () => {
+    const coder = new BigNumberCoder('u64');
+    const value: number = 76472027892439376;
+
+    expect(() => coder.encode(value)).toThrow(
+      new FuelError(ErrorCode.ENCODE_ERROR, 'Invalid u64 type - number value is too large.')
```

Comment:
```suggestion
      new FuelError(ErrorCode.ENCODE_ERROR, 'Invalid u64 type - too large. Number can only safely handle up to 53 bits').
```
 JW if it needs more detail